src/main/java/com/efficio/fieldbook/web/common/controller/CrossingSettingsController.java

```diff
@@ -369,6 +370,14 @@ boolean checkForHybridMethods(final List<ImportedCross> importedCrosses) {
 		return this.crossExpansionProperties.getHybridBreedingMethods();
 	}
 
+	@ResponseBody
+	@RequestMapping(value = "/getLocationIdFromFirstEnviroment", method = RequestMethod.GET)
+	public String getLocationIdFromEnviroment() {
+		return this.studySelection.getWorkbook().getTrialObservations().get(0).getDataList().stream()
+			.filter(measurementData -> measurementData.getMeasurementVariable().getTermId() == TermId.LOCATION_ID.getId()).findFirst().get()
+			.getValue();
+	}
+
```

Comment:
Is it possible to use one of the pre-existing API's? like `/{cropname}/programs/{programUUID}/studies/{studyId}/instances`

Reply:
As discussed, this is fine for now, we'll work more in this area in the future.